libglusterfs/src/glusterfs/gf-dirent.h

```diff
@@ -53,6 +53,13 @@ struct _gf_dirent {
 
 #define DT_ISDIR(mode) (mode == DT_DIR)
 
+static inline gf_boolean_t
+inode_dir_or_parentdir(gf_dirent_t *entry)
+{
+    return (entry->d_len <= 2 && (strcmp(entry->d_name, ".") == 0 ||
+                                  strcmp(entry->d_name, "..") == 0));
```

Comment:
We can avoid the `strcmp()`:

```suggestion
    return (entry->d_len <= 2) && (entry->d_name[0] == '.') &&
           ((entry->d_name[1] == 0) || (entry->d_name[1] == '.'));
```

Another option is to just check the `d_name` (in this case we don't need the full `gf_dirent_t`, just a `char *` and could also be used for non-entry structures:

```suggestion
    return (name[0] == '.') &&
           ((name[1] == 0) || ((name[1] == '.') && (name[2] == 0)));
```

Reply:
OK, implemented the 1st suggestion (without d_len check?)